### PR TITLE
fix(telemetry): display custom event attributes properly on subscription events

### PR DESCRIPTION
**Pull request by bnjjj**
## Display custom event attributes properly on subscription events

Custom event attributes set using selectors at the supergraph level is now displayed properly. Example of configuration:

```yaml
telemetry:
  instrumentation:
    events:
      supergraph:
        supergraph.event:
          message: supergraph event
          on: event_response # on every supergraph event (like subscription event for example)
          level: info
          attributes:
            test:
              static: foo
            response.data:
              response_data: $ # Display all the response data payload
            response.errors:
              response_errors: $ # Display all the response errors payload
```

## Don't send `apollo_private.*` attributes to  Jaeger collector

When using Jaeger collector to send traces you will no longer receive useless span attributes prefixed with `apollo_private.`

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [x] Changes are compatible[^1]
- [x] Documentation[^2] completed
- [x] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [x] Integration Tests
    - [x] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

File: .changesets/fix_latte_lens_tent_blaze.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+### Don't send `apollo_private.*` attributes to  Jaeger collector ([PR #6033](https://github.com/apollographql/router/pull/6033))
+
+
+When using Jaeger collector to send traces you will no longer receive useless span attributes prefixed with `apollo_private.`
```
Comment (lrlna) [inline .changesets/fix_latte_lens_tent_blaze.md:4]:
```suggestion
When using Jaeger collector to send traces you will no longer receive unnecessary span attributes prefixed with `apollo_private.`
```

File: apollo-router/src/plugins/telemetry/config_new/events.rs [hunk 2]
```diff
@@ -738,7 +739,10 @@ where
             let mut new_attributes = selectors.on_response_event(response, ctx);
             attributes.append(&mut new_attributes);
         }
-
+        // Dumb span to make sure the custom attributes are saved in current span extensions
```
Comment (lrlna) [inline apollo-router/src/plugins/telemetry/config_new/events.rs:742]:
```suggestion
        // Stub span to make sure the custom attributes are saved in current span extensions
```

File: .changesets/fix_latte_lens_tent_blaze.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+### Don't send `apollo_private.*` attributes to  Jaeger collector ([PR #6033](https://github.com/apollographql/router/pull/6033))
+
```
Comment (abernix) [inline .changesets/fix_latte_lens_tent_blaze.md:2]:
```suggestion
### Internal `apollo_private.*` attributes are not sent to the Jaeger collector ([PR #6033](https://github.com/apollographql/router/pull/6033))
```

**Review by abernix — COMMENTED** [review]
It looks like you added some tests but only checked the "manual tests" test checkbox at the top.  Is that an oversight, or do you feel like it needs different tests.

File: apollo-router/src/plugins/telemetry/dynamic_attribute.rs [hunk 4]
```diff
@@ -240,7 +240,9 @@ impl EventDynAttribute for ::tracing::Span {
         self.with_subscriber(move |(id, dispatch)| {
             if let Some(reg) = dispatch.downcast_ref::<Registry>() {
                 match reg.span(id) {
-                    None => eprintln!("no spanref, this is a bug"),
+                    None => {
+                        eprintln!("no spanref, this is a bug");
```
Comment (BrynCooke) [inline apollo-router/src/plugins/telemetry/dynamic_attribute.rs:244]: This should panic in debug builds?

Comment (bnjjj) [reply apollo-router/src/plugins/telemetry/dynamic_attribute.rs:244]: I'll do this in a follow up or as it's not a change

**Pull request merged**